zetaclient/chains/solana/observer/outbound.go

```diff
@@ -109,6 +109,9 @@ func (ob *Observer) VoteOutboundIfConfirmed(ctx context.Context, cctx *crosschai
 
 	// status was already verified as successful in CheckFinalizedTx
 	outboundStatus := chains.ReceiveStatus_success
+	if inst.Failed() {
```

Comment:
@skosito is it possible that someone who makes a withdraw tx and fails because no signature, and this causes a revert?

Reply:
Since this gateway contract is open anyone can call it (and fail for invalid signature), 
in this case the failed tx should not be counted as failed outbound, rather it should just be ignored. 

> but i guess we can also skip revert in the code same as NonceMismatch error?

NonceMismatch and invalid signature I think. Basically any error other than the CPI call failure.

Reply:
In the `CheckFinalizedTx` method, failed tx (due to whatever error) is skipped by below logic
```
// the tx must be successful in order to effectively increment the nonce
if txResult.Meta.Err != nil {
    logger.Error().Any("Err", txResult.Meta.Err).Msg("tx is not successful")
    return nil, false
}
```

So, all remaining txs that arrive the call to [ParseGatewayInstruction](https://github.com/zeta-chain/node/blob/sol-withdraw-and-call-revert/zetaclient/chains/solana/observer/outbound.go#L251) are valid (status is success) outbounds carrying valid `TSS` signatures.